### PR TITLE
Core/Loading: output error when specified emote id for some quest-related tables is invalid

### PR DESCRIPTION
**Pull request by SnapperRy**
**Changes proposed**: title says it all.

**Target branch(es)**: 335

**Tests performed**:  Compiles and works.

### CODE REVIEW

File: src/server/game/Quests/QuestDef.cpp [hunk 1]
```diff
@@ -142,13 +150,25 @@ void Quest::LoadQuestRequestItems(Field* fields)
...
     RequestItemsText = fields[3].GetString();
 }
 
 void Quest::LoadQuestOfferReward(Field* fields)
 {
     for (int i = 0; i < QUEST_EMOTE_COUNT; ++i)
+    {
+        if (fields[1+i].GetUInt16() >= EMOTE_MAX)
+        {
+            TC_LOG_ERROR("sql.sql", "Table `quest_offer_reward` has non-existing Emote%u %u set for quest %u. Skipped.", 1+i, fields[1+i].GetUInt16(), fields[0].GetUInt32());
```
Comment (Carbenium) [inline src/server/game/Quests/QuestDef.cpp]: Same as above

File: src/server/game/Quests/QuestDef.cpp [hunk 2]
```diff
@@ -132,7 +132,15 @@ Quest::Quest(Field* questRecord)
 void Quest::LoadQuestDetails(Field* fields)
 {
     for (int i = 0; i < QUEST_EMOTE_COUNT; ++i)
+    {
+        if (fields[1+i].GetUInt16() >= EMOTE_MAX)
```
Comment (Shauren) [inline src/server/game/Quests/QuestDef.cpp]: Don't define EMOTE_MAX and don't check it like this
Verify the emote exists in Emotes.dbc instead (we load it, lookup sEmotesStore)

File: src/server/game/Quests/QuestDef.cpp [hunk 3]
```diff
@@ -132,7 +132,15 @@ Quest::Quest(Field* questRecord)
...
+        {
+            TC_LOG_ERROR("sql.sql", "Table `quest_details` has non-existing Emote%u %u set for quest %u. Skipped.", 1+i, fields[1+i].GetUInt16(), fields[0].GetUInt32());
```
Comment (jackpoz) [inline src/server/game/Quests/QuestDef.cpp]: why is %i suggested for uint16() ?

Comment (Carbenium) [inline src/server/game/Quests/QuestDef.cpp]: Because the variable ```i``` is an int

Comment (jackpoz) [inline src/server/game/Quests/QuestDef.cpp]: lul TC code is not even consistent about `%i` and `%d`

**Pull request merged**